Feature/endpoints

pullveryzator opened the pull request:
Реализовал часть эндпоинтов. Пока нет юзеров. С ними возникли некоторые сложности. Подробнее опишу в пачке. Немного поравил импорт в app/core/db.py. Добавил роутеры и подключил их в main.py. Приложение запускал командой python main.py

pullveryzator committed:
First commit for endpoints
Changes to be committed:
	new file:   src/app/core/dependencies.py
	modified:   src/app/endpoints/__init__.py
	new file:   src/app/endpoints/group.py
	new file:   src/app/endpoints/message.py
	new file:   src/app/endpoints/photo.py
	new file:   src/app/endpoints/routers.py
	new file:   src/app/endpoints/user_tg.py
	new file:   src/app/schemas/group.py
	new file:   src/app/schemas/message.py
	new file:   src/app/schemas/photo.py
	new file:   src/app/schemas/user_tg.py

pullveryzator committed:
Реализовал часть эндпоинтов. Пока нет юзеров. С ними возникли некоторые сложности. Подробнее опишу в пачке.
Немного поравил импорт в app/core/db.py.
Добавил роутеры и подключил их в main.py.
Приложение запускал командой python main.py

On branch feature/endpoints
Changes to be committed:
	modified:   app/core/db.py
	deleted:    app/core/dependencies.py
	modified:   app/endpoints/__init__.py
	modified:   app/endpoints/group.py
	modified:   app/endpoints/message.py
	modified:   app/endpoints/photo.py
	modified:   app/endpoints/routers.py
	modified:   app/schemas/group.py
	modified:   app/schemas/message.py
	modified:   main.py

Changes not staged for commit:
	modified:   app/crud/base.py
	modified:   app/crud/user_tg.py
	modified:   app/endpoints/user_tg.py
	modified:   app/schemas/user_tg.py

Untracked files:
	app/templates/

KonstantinRaikhert commented on src/app/core/db.py:
```diff
@@ -4,7 +4,7 @@
 from sqlalchemy.orm import declarative_base, declared_attr, sessionmaker
 
-from src.bot.core.settings import settings
+from bot.core.settings import settings
```
Кажется на проекте сейчас проблема с импортами. Кто то удаляет кто то добавляет)
Необходимо разрешить проблему. На свякий случай проверить везде __init__.py

KonstantinRaikhert commented on src/app/endpoints/group.py:
Нужно как-то обозначить в структуре файлов, что это ендпоинты для админки.

pullveryzator replied:
Переименовал папку endpoints.

pullveryzator committed:
Fix user_tg and schemas
Changes to be committed:
	modified:   app/endpoints/user_tg.py
	modified:   app/schemas/message.py
	modified:   app/schemas/user_tg.py

Changes not staged for commit:
	modified:   app/crud/base.py
	modified:   app/crud/user_tg.py

Untracked files:
	app/templates/

pullveryzator committed:
Доделал эндпоинты юзеров. Нужно обязательно поправить модели.

Changes to be committed:
	new file:   __init__.py
	new file:   group.py
	new file:   message.py
	new file:   photo.py
	new file:   routers.py
	new file:   user_tg.py
	modified:   ../../main.py

Changes not staged for commit:
	modified:   ../core/db.py
	modified:   ../crud/base.py
	modified:   ../crud/user_tg.py
	deleted:    ../endpoints/__init__.py
	deleted:    ../endpoints/group.py
	deleted:    ../endpoints/message.py
	deleted:    ../endpoints/photo.py
	deleted:    ../endpoints/routers.py
	deleted:    ../endpoints/user_tg.py
	modified:   ../models/models.py

Untracked files:
	../../alembic/versions/2024_11_29_1555-01c576f4b80a_remove_id_field_from_user_group_table.py
	../templates/

pullveryzator committed:
PEP8 in schema/user_tg.py

On branch feature/endpoints
Your branch is up to date with 'origin/feature/endpoints'.

Changes to be committed:
	deleted:    endpoints/__init__.py
	deleted:    endpoints/group.py
	deleted:    endpoints/message.py
	deleted:    endpoints/photo.py
	deleted:    endpoints/routers.py
	deleted:    endpoints/user_tg.py
	modified:   schemas/user_tg.py

Changes not staged for commit:
	modified:   core/db.py
	modified:   crud/base.py
	modified:   crud/user_tg.py
	modified:   models/models.py

Untracked files:
	../alembic/versions/2024_11_29_1555-01c576f4b80a_remove_id_field_from_user_group_table.py
	templates/

pullveryzator commented:
Добавил юзеров. В другой ветке поправлю круды для юзеров.

pullveryzator committed:
Доработка контекста для шаблона.
On branch feature/endpoints
Your branch is up to date with 'origin/feature/endpoints'.

Changes to be committed:
	modified:   app/admin_endpoints/user_tg.py

Changes not staged for commit:
	modified:   app/admin_endpoints/routers.py
	modified:   app/core/db.py
	modified:   app/crud/base.py
	modified:   app/crud/user_tg.py
	modified:   app/models/models.py

Untracked files:
	alembic/versions/2024_11_29_1555-01c576f4b80a_remove_id_field_from_user_group_table.py
	app/admin_endpoints/auth.py
	app/templates/

pullveryzator committed:
Merge branch 'develop' of github.com:Studio-Yandex-Practicum/providenie into feature/endpoints

pullveryzator committed:
Fix all endpoints.

On branch feature/endpoints
Your branch is ahead of 'origin/feature/endpoints' by 20 commits.
  (use "git push" to publish your local commits)

Changes to be committed:
	new file:   app/admin_endpoints/dashboard.py
	modified:   app/admin_endpoints/group.py
	modified:   app/admin_endpoints/message.py
	modified:   app/admin_endpoints/routers.py
	modified:   app/admin_endpoints/user_tg.py
	modified:   app/schemas/user_tg.py

Changes not staged for commit:
	modified:   app/core/db.py
	modified:   app/crud/base.py
	modified:   app/crud/user_tg.py
	modified:   app/models/models.py
	modified:   main.py

Untracked files:
	alembic/versions/2024_11_29_1555-01c576f4b80a_remove_id_field_from_user_group_table.py
	app/admin_endpoints/static/
	app/templates/

pullveryzator committed:
Merge branch 'develop' of github.com:Studio-Yandex-Practicum/providenie into feature/endpoints

pullveryzator committed:
Добавил поле tg_id в шаблон. Основная логика работает (список пользователей, групп, сообщений, создание/редактирование/удаление/ сообщений, групп. Создание пользователя). В коде эндпоинта юзера переименовал переменную для совместимостью с шаблоном. Не работает сохранение фотографий. Ошибка: отказано в доступе записи.

On branch feature/endpoints
Your branch is ahead of 'origin/feature/endpoints' by 84 commits.
  (use "git push" to publish your local commits)

Changes to be committed:
	new file:   src/alembic/versions/2024_12_05_1805-5aeecc3b5b62_remove_id_field_from_user_group_.py
	modified:   src/app/admin_endpoints/group.py
	modified:   src/app/admin_endpoints/message.py
	modified:   src/app/admin_endpoints/photo.py
	new file:   src/app/admin_endpoints/static/css/styles.css
	modified:   src/app/admin_endpoints/user_tg.py
	modified:   src/app/core/db.py
	modified:   src/app/crud/base.py
	modified:   src/app/crud/user_tg.py
	modified:   src/app/models/models.py
	new file:   src/app/templates/__init__.py
	modified:   src/app/templates/create_user.html
	new file:   src/app/templates/group_created.html
	new file:   src/app/templates/group_deleted.html
	new file:   src/app/templates/message_created.html
	new file:   src/app/templates/message_deleted.html
	new file:   src/app/templates/message_updated.html
	new file:   src/app/templates/user_created.html
	new file:   src/app/templates/user_updated.html
	modified:   src/main.py

teamofroman committed:
Rebase to fix/global_fix

teamofroman committed:
Исправления, добавки

teamofroman committed:
Исправлены импорты

pullveryzator committed:
Merge branch 'develop' of github.com:Studio-Yandex-Practicum/providenie into feature/endpoints

pullveryzator committed:
Поправлены шаблоны.
Переопределен метод create в crud_message.
Допилены эндпоинты и схемы.
Добавлен проперти в модель для корректной подгрузки статики.
Добавил зависимости в эндпоинты, но пока оставил закомментированными.
Много мелких правок.

On branch feature/endpoints
Your branch is ahead of 'origin/feature/endpoints' by 13 commits.
  (use "git push" to publish your local commits)

Changes to be committed:
	new file:   app/admin_endpoints/constants.py
	modified:   app/admin_endpoints/dashboard.py
	modified:   app/admin_endpoints/group.py
	modified:   app/admin_endpoints/message.py
	modified:   app/admin_endpoints/photo.py
	modified:   app/admin_endpoints/user_tg.py
	modified:   app/crud/base.py
	modified:   app/crud/message.py
	modified:   app/crud/user_tg.py
	modified:   app/models/constants.py
	modified:   app/models/models.py
	modified:   app/schemas/message.py
	modified:   app/schemas/user_tg.py
	modified:   app/templates/admin_dashboard.html
	modified:   app/templates/admin_groups.html
	modified:   app/templates/admin_messages.html
	modified:   app/templates/admin_users.html
	modified:   app/templates/create_group.html
	modified:   app/templates/create_message.html
	modified:   app/templates/create_user.html
	modified:   app/templates/edit_group.html
	modified:   app/templates/edit_message.html
	modified:   app/templates/edit_user.html
	modified:   app/templates/footer.html

Changes not staged for commit:
	modified:   app/routers/auth.py

Untracked files:
	app/admin_endpoints/static/photos/

pullveryzator committed:
Подключил аунтификацию и регистрацию. На эндпоинте регистрации стоит depends.
Чтобы создать первого юзера нужно снять депендс.
Доработка шаблонов, эндпоинтов и небольшая правка схем обновления сообщений.
Внес правки в механизм генерации декодирования токена по согласованию с техлидом.
Обьединил файлы main.py.
Подтянул из ветки девелоп файл docker-compose.
Куча мелких правок...

On branch feature/endpoints
Your branch is up to date with 'origin/feature/endpoints'.

Changes to be committed:
	modified:   ../infra/docker-compose.yaml
	modified:   app/admin_endpoints/constants.py
	modified:   app/admin_endpoints/dashboard.py
	modified:   app/admin_endpoints/group.py
	modified:   app/admin_endpoints/message.py
	modified:   app/admin_endpoints/photo.py
	modified:   app/admin_endpoints/user_tg.py
	modified:   app/core/authentication.py
	modified:   app/core/jwt.py
	deleted:    app/main.py
	modified:   app/routers/auth.py
	modified:   app/schemas/message.py
	renamed:    app/admin_endpoints/static/css/bootstrap.min.css -> app/static/css/bootstrap.min.css
	renamed:    app/admin_endpoints/static/css/styles.css -> app/static/css/styles.css
	renamed:    app/admin_endpoints/static/js/bootstrap.bundle.min.js -> app/static/js/bootstrap.bundle.min.js
	new file:   app/templates/403.html
	modified:   app/templates/404.html
	modified:   app/templates/admin_groups.html
	modified:   app/templates/admin_messages.html
	modified:   app/templates/admin_sidebar.html
	modified:   app/templates/admin_users.html
	modified:   app/templates/create_message.html
	modified:   main.py

Untracked files:
	app/static/photos/

pullveryzator committed:
Визуальная правка шаблона редактирования пользователя.

On branch feature/endpoints
Your branch is up to date with 'origin/feature/endpoints'.

Changes to be committed:
	modified:   app/templates/edit_user.html

Untracked files:
	app/static/photos/

pullveryzator committed:
1. Поле ввода пароля в шаблоне edit_user сделал необязательным.
   Написал валидаторы для схемы обновления + поправил круд обновления пользователя, чтобы пустой пароль не попал в базу.
2. Добавил зависимость в стартовую страницу.
3. При ошибке аунтификации срабатывает исключение, которое перехватывает хендлер 401.
   Далее он переадресовывает на страницу входа с выводом текста ошибки из поля detail.
4. Добавил хендлер для 403 ошибки.
5. Добавить хендлер для 500 ошибки.
6. Добавил хендлер для общих ошибок (в самом конце main.py)

On branch feature/endpoints
Your branch is up to date with 'origin/feature/endpoints'.

Changes to be committed:
	modified:   app/admin_endpoints/group.py
	modified:   app/admin_endpoints/message.py
	modified:   app/admin_endpoints/photo.py
	modified:   app/admin_endpoints/user_tg.py
	modified:   app/core/auth.py
	modified:   app/crud/user_tg.py
	modified:   app/routers/auth.py
	modified:   app/schemas/user_tg.py
	new file:   app/templates/400.html
	modified:   app/templates/401.html
	modified:   app/templates/403.html
	modified:   app/templates/404.html
	new file:   app/templates/500.html
	modified:   app/templates/admin_messages.html
	modified:   app/templates/edit_user.html
	modified:   main.py

Untracked files:
	app/static/photos/

pullveryzator committed:
Мелкие правки перевода.

On branch feature/endpoints
Your branch is ahead of 'origin/feature/endpoints' by 1 commit.
  (use "git push" to publish your local commits)

Changes to be committed:
	modified:   app/core/auth.py

Untracked files:
	app/static/photos/

pullveryzator committed:
On branch feature/endpoints
Your branch is up to date with 'origin/feature/endpoints'.

Changes to be committed:
	modified:   app/admin_endpoints/user_tg.py

Changes not staged for commit:
	modified:   ../infra/.env.example

Untracked files:
	app/static/photos/

pullveryzator committed:
Проверка на передачу пустого пароля в эндпоинте редактирования пользователя.

On branch feature/endpoints
Your branch is ahead of 'origin/feature/endpoints' by 1 commit.
  (use "git push" to publish your local commits)

Changes to be committed:
	modified:   app/admin_endpoints/user_tg.py
	modified:   app/schemas/user_tg.py

Changes not staged for commit:
	modified:   ../infra/.env.example

Untracked files:
	app/static/photos/

teamofroman reviewed (approved):
Отлично
